### PR TITLE
feat(collaboration): add grant permission

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 1]
```diff
@@ -1898,4 +1959,47 @@ export class TeamsAppSolution implements Solution {
...
+    }
+    const tenantId = currentUser["tid"] as string;
+
+    if (email) {
+      const graphToken = await ctx.graphTokenProvider?.getAccessToken();
+      const instance = axios.create({
+        baseURL: "https://graph.microsoft.com/v1.0",
+      });
+      instance.defaults.headers.common["Authorization"] = `Bearer ${graphToken}`;
+      const res = await instance.get(`/users`);
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1976]: This request may only return first 100 users, you need to user filter to get user

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts]: Fixed. Thx!

File: packages/fx-core/src/plugins/solution/fx-solution/question.ts [hunk 2]
```diff
@@ -224,3 +225,9 @@ export const ProgrammingLanguageQuestion: SingleSelectQuestion = {
     return "Select a programming language.";
   },
 };
+
+export const GrantPermissionQuestion: TextInputQuestion = {
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/question.ts:229]: We should also consider to ask user to select environment

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/question.ts]: Fixed. Thanks.

**Review by SLdragon — APPROVED** [review]

**Pull request merged**